package ovp.backend.controller;

import java.util.List;

import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.PUT;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import org.codehaus.jettison.json.JSONException;
import org.codehaus.jettison.json.JSONObject;
import org.springframework.context.support.ClassPathXmlApplicationContext;

import ovp.backend.persistence.dao.UserDAO;
import ovp.backend.persistence.model.User;

@Path("/users")
public class UserController {
	ClassPathXmlApplicationContext context = new ClassPathXmlApplicationContext("spring.xml");
	
	UserDAO userDAO = context.getBean(UserDAO.class);
	
	@GET
	@Produces(MediaType.APPLICATION_JSON)
	public List<User> getUsers(){
		return userDAO.displayUsers();
	}
	
	@GET
	@Path("/{userName}")
	@Produces(MediaType.APPLICATION_JSON)	
	public User findUser(@PathParam("userName") String userName) {
		return userDAO.findUserByUserName(userName);
	}
	
	@POST 
	@Produces(MediaType.APPLICATION_JSON)
	public void addUser(User user){
		userDAO.createUser(user);		
	}

	@PUT
	public void updateUser(String json) {
		
		JSONObject jsonObj;
		try {
			jsonObj = new JSONObject(json);
			String userName = jsonObj.getString("userName");
			String password = jsonObj.getString("password");
			userDAO.updateUser(userName, password);
		} catch (JSONException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}
	
}





















